uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -478,6 +478,30 @@ public static function canSeeKeyDates($contactId) {
   }
 
   /**
+   * Check if given Contact ID has access to required permission to create
+   * activityURL.
+   *
+   * @param int $contactId
+   * @return string
+   */
+  public static function createActivityURL($contactId = NULL) {
+
+    if ($contactId) {
+      $activityUrl = '';
+      $drupal_user = user_load(_get_uf_match_contact($contactId)['uf_id']);
+
+      if (user_access('access CiviCRM', $drupal_user)) {
+        $activityUrl = CIVICRM_UF_BASEURL . '/civicrm/activity/view?action=view&reset=1&id=' . $activityResult->id . '&cid=' . $contactId . '&context=activity&searchContext=activity';
```

Comment:
If you want to use $activityResult here then you need to pass it into the function on line 412.